### PR TITLE
task/refactor_geo_message/send_geo_vm

### PR DESCRIPTION
**Pull request by teamagam1**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/105" target="_blank">ReviewNinja

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/gestures/GGMapGestureListener.java [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
 import com.teamagam.gimelgimel.app.map.view.GGMap;
 import com.teamagam.gimelgimel.domain.base.logging.Logger;
 import com.teamagam.gimelgimel.app.common.logging.LoggerFactory;
 import com.teamagam.gimelgimel.app.utils.Constants;
-import com.teamagam.gimelgimel.app.view.fragments.dialogs.SendGeographicMessageDialog;
+import com.teamagam.gimelgimel.app.message.view.SendGeographicMessageDialog;
 import com.teamagam.gimelgimel.app.map.model.geometries.PointGeometry;
 
 public class GGMapGestureListener extends SimpleOnMapGestureListener {
```
Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/gestures/GGMapGestureListener.java:10]: it implements SimpleOnMapGestureListener , which is redundant

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/IMapView.java [hunk 2]
```diff
@@ -13,4 +13,6 @@
     void goToLocation(PointGeometry location);
 
     void addLayer(VectorLayer vectorLayer);
+
+    void openSendGeoDialog(PointGeometry pointGeometry);
```
Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/IMapView.java:17]: i am not sure take picture and gotolocatin should be here.
they are not really connected to the map

File: app/src/main/java/com/teamagam/gimelgimel/app/injectors/components/MessagesComponent.java [hunk 3]
```diff
@@ -2,13 +2,15 @@
...
+import com.teamagam.gimelgimel.app.message.view.SendGeographicMessageDialog;
 import com.teamagam.gimelgimel.app.view.fragments.dialogs.SendMessageDialogFragment;
+import com.teamagam.gimelgimel.presentation.scopes.PerActivity;
+import com.teamagam.gimelgimel.presentation.scopes.PerFragment;
 
 import dagger.Component;
 
 /**
- * A scope {@link PerFragment} component.
+ * A scope {@link PerActivity} component.
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/injectors/components/MessagesComponent.java:13]: PerActivity, but the annotation is still PerFragment.

File: app/src/main/java/com/teamagam/gimelgimel/app/injectors/modules/ApplicationModule.java [hunk 4]
```diff
@@ -36,12 +38,18 @@ Context provideApplicationContext() {
...
     @Singleton
-    MessagesRepository provideUserRepository(MessagesDataRepository messageRepo) {
+    MessagesRepository provideMessageRepository(MessagesDataRepository messageRepo) {
         return messageRepo;
     }
 
     @Provides
     @Singleton
+    GeoEntityRepository provideGeoRepository(GeoDataRepository geoRepo) {
+        return geoRepo;
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/injectors/modules/ApplicationModule.java:48]: Remove @inject from constructor and declare the variable here.

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/gestures/OnMapGestureListener.java [hunk 5]
```diff
@@ -1,18 +1,22 @@
...
+package com.teamagam.gimelgimel.app.map.viewModel.gestures;
 
 import com.teamagam.gimelgimel.app.map.model.geometries.PointGeometry;
 
 /**
  * TODO: add class summary notes
  */
 public interface OnMapGestureListener {
+
+    @SuppressWarnings("unused")
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/gestures/OnMapGestureListener.java:10]: Remove the warning supress, it's not necessary.

File: app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/Message.java [hunk 6]
```diff
@@ -22,7 +22,7 @@
     public static final String USER_LOCATION = "UserLocation";
     public static final String IMAGE = "Image";
     @SerializedName("content")
-    protected T mContent;
+    protected T content;
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/Message.java:25]: Why not mContent?
It's still a member.

File: app/src/main/java/com/teamagam/gimelgimel/app/message/view/SendGeographicMessageDialog.java [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+    }
+
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+        DaggerMessagesComponent.builder()
+                .applicationComponent(((GGApplication) getActivity().getApplication()).getApplicationComponent())
+                .messageModule(new MessageModule())
+                .build()
+                .inject(this);}
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/message/view/SendGeographicMessageDialog.java:83]: Add line to the bracket.

File: app/src/main/java/com/teamagam/gimelgimel/app/message/view/SendGeographicMessageDialog.java [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+    @Override
+    protected boolean hasPositiveButton() {
+        return true;
+    }
+
+    @Override
+    protected void onPositiveClick() {
+        sLogger.userInteraction("Clicked OK");
+        mViewModel.clickedOK();
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/message/view/SendGeographicMessageDialog.java:145]: I like to call the method after what they do.
We'll change it because of our discussion, but "clickedOK()" isn't a good name.
Rename it to what the method does, because we know the user clicked ok (The overriden method name).

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/SendGeoMessageViewModel.java [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+        super();
+    }
+
+    public void init(ISendGeoMessageView view, PointGeometry point) {
+        this.types = context.getResources().getStringArray(R.array.geo_locations_types);
+        this.point = point;
+        this.mView = view;
+    }
+
+    public void clickedOK() {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/SendGeoMessageViewModel.java:53]: See the comment in the DialogFragment class.

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/SendGeoMessageViewModel.java [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+        if (isInputValid()) {
+            mPresenter.sendGeoMessage(GGMessageSender.getUserName(context), text, point.latitude,
+                    point.longitude, point.altitude, types[typeIdx]);
+            mView.dismiss();
+        } else {
+            //validate that the user has entered description
+            sLogger.userInteraction("Input not valid");
+            mView.showError();
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/SendGeoMessageViewModel.java:61]: Show error with string?

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/SendGeoMessageViewModel.java [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+    public void setTypeIdx(int type) {
+        this.typeIdx = type;
+    }
+
+    public int getTypeIdx() {
+        return typeIdx;
+    }
+
+    public interface ISendGeoMessageView {
+        void showError();
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/SendGeoMessageViewModel.java:91]: Show error should get string in the parameters.

File: app/src/main/java/com/teamagam/gimelgimel/app/model/entities/GeoContent.java [hunk 12]
```diff
@@ -9,41 +9,41 @@
 public class GeoContent {
 
     @SerializedName("location")
-    private PointGeometry mPoint;
+    private PointGeometry point;
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/model/entities/GeoContent.java:12]: Add lower camel case "m" to all the members here.
Why did you change it?

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/base/BaseBindingDialogFragment.java [hunk 13]
```diff
@@ -0,0 +1,268 @@
...
+                        //Do nothing!
+                    }
+                };
+
+        // Set buttons. A listener must be passed for the dialog to construct the buttons
+        // Builder will add the listener to button's click listening pipe that will eventually
+        // dismiss the button on every click. To overcome it, we override the behaviour on OnStart
+        // after the builder finished building the dialog
+        if (hasPositiveButton()) {
+            builder.setPositiveButton(getPositiveString(), doNothingListener);
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/base/BaseBindingDialogFragment.java:88]: Why not set the listeners here?
You add them later, so add them here instread.

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/IMapView.java (hunk 2 above)
Comment (teamagam1) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/IMapView.java:17]: they are here until we will correct when we will get to the feature
